models/transaction.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * This file contains wrapper functions to interact with the DB.
+ * This will contain the DB schema if we start consuming an ORM for managing the DB operations
+ */
+
+const firestore = require('../utils/firestore')
+const transactionsModel = firestore.collection('transaction')
+
+/**
+ * fetch latest N transactions from transactions collection for specific userid provided
+ *
+ * @param userId { String }: User Id String to be used to fetch latest transactions
+ * @param limit { number }: no of records to fetch from transaction table, if not specified in URL then default will be 10
+ * @param offset { number }: starting index of set of records, default value is 0, useful for pagination
+ * @param orderBy { String }: to order the transactions in ascending or descending manner, by default will fetch latest first
+ * @returns {Promise<{transactions: []}>}
+ */
+const fetch = async (userId, offset, limit, orderBy) => {
```

Comment:
Better to name this function as `fetchTransactions`? 🤔

Reply:
fetch is defined under transaction controller, it's means fetching the transactions

Reply:
We already have a util function called `fetch` which is used for network calls. That might become confusing. Also if I use your function in some other file and I destructure it there, it won't make any sense as to what the function is doing there.. I hope that makes sense

Reply:
For the functions like (toString or fetch), same name is suggested and to distinguish we have documentation. User can't remember diff names for same kind of task related to diff APIs.

Reply:
> Also if I use your function in some other file and I destructure it there, it won't make any sense as to what the function is doing there.. I hope that makes sense

Say I use this function in `models/wallets.js`, and I write `fetch(.....)` how do you know what is being fetched? The function names should be self-explanatory. You shouldn't need to look at the imports to know where the file is coming from.

Imagine all the places we replace `fetchUser => fetch`, `fetchTransactions => fetch` and if I had to use both the functions in one single file, how would I differentiate?

Reply:
For that we have documentation. `fetch(..)` is like atomic operation which every API supposed to have, if you provide different name for each then you will end up with 1000s of method name.

Reply:
You didn't say anything about:
> Imagine all the places we replace `fetchUser => fetch`, `fetchTransactions => fetch` and if I had to use both the functions in one single file, how would I differentiate?

Reply:
every fetch is followed an entity like user or transaction, if you are creating object for that entity the you should be aware of object name(keep object name in that way). At last we have documentation in place for every help required.

Reply:
I'm talking about function names themselves, not the objects returned by them

Also, how do you know that this is going to be the only `fetch` function in this file? Can we not have `fetchTransaction` to get one particular transaction details given it's id tomorrow? What would you do that in that case?

We need to differentiate between `fetchTransactions` and `fetchTransaction`

Reply:
You can fetch any number of transactions with this API but value of limit is required for that otherwise it will return only 10 maximum records

Reply:
Hey @rkmr039 , Even I agree with what @swarajpure is mentioning here, even though the function is defined inside models/transaction , it will create confusions when we try to import fetch from general util and usually function names are self explanatory...

Reply:
This function will fetch the transactions by a user. I'm talking about a function which would return transaction details given transaction id

Reply:
With name fetchTransactions , we are defining a function in general which has a capability to fetch N transactions 0<=N<=infinity

I hope I am describing it right

Reply:
fetchTransactions/: transactionId can provide unique transaction details
Pardon for some spelling mistakes as I am using mobile to comment here

Reply:
In fact, the name shouldn't be just `fetchTransactions`, it should be `fetchTransactionsByUser`

Tomorrow we could have:
- `fetchAllTransactions`: Returns ALL of the transactions in the DB (by all users)
- `fetchTransactionById`: Returns a transaction by transactionId

So let's go with `fetchTransactionsByUser` @rkmr039

Reply:
Regarding the 1000s of method names, would you prefer 1000 functions with different names (self-explanatory) or 1000 functions with same name (all called `fetch`)? 😉

Reply:
Every function performs same kind of task can have same name although in diff packages.

Reply:
renamed as `fetchByUsername`

Reply:
@swarajpure @rkmr039 great discussion guys, we have come to collective conclusion now

Reply:
Actually I always want to be a part of team like this, but in my current role i can name methods anything ....nobody checks

Reply:
> Every function performs same kind of task can have same name although in diff packages.

Why are you not getting the point?
```JS
const { fetch } = require('./models/transactions')
const { fetch } = require('./models/users')
```

Is this the confusion you would like to work with?

Every function HAS to be named as per the job it is supposed to do irrespective of the folder/file name. If the function was named `fetchTransactionsModel` then keep the `model` in function name makes no sense, but `transactions` MUST be kept. 

This function may not remain exclusive to this route. We may use this in other routes as well. Do you want the developer to look at the imports every single time to make sense of what the function is supposed to do? That defeats the purpose of having self-explanatory function names in the first place. 

Why not just name all the functions as `a`? Short and sweet.

Reply:
> renamed as `fetchByUsername`

Even this doesn't make much sense.
Why should I have to look at the file name to make sense of what this function fetches? 
Why should I need to look at documentation if having self-explanatory function name could solve the problem?

Reply:
Do you want me to rename it as `fetchTransactionByUsername` ?

Reply:
`fetchTransactionByUserId` in this case

Reply:
Yeah, fetchTransactionsByUserId would be better in this case right 
And the name remains same in model as well as in controller for easy navigation to different functions

Reply:
Not exactly the same, the controller is taking `username` and the model is taking `userId`, but they'll be similar, yeah

Reply:
Even though it is taking username in query parama , it is used to fetch userId and then the model operation occurs right... My implications was in that fashion

Reply:
Done